lib/SensorFusion/src/ExtendedKalmanFilter.cpp

```diff
@@ -54,51 +54,119 @@
+ *  The factor 0.025 respresents the time period of 25 ms. */
+constexpr float TURNRATE_VARIANCE = 4.0F * 4.0F * 0.025F * 0.025F;
+
+/** Variance of the Odometry Position. Determined empirically. */
+constexpr float ODOMETRY_POSITION_VARIANCE = 2.0F / 1000.0F;
+
+/** Variance of the Odometry Velocity. Determined empirically. */
+constexpr float ODOMETRY_VELOCITY_VARIANCE = 3.0F / 1000.0F;
+
+/** Variance of the Odometry Angle. Determined empirically. */
+constexpr float ODOMETRY_ANGLE_VARIANCE = 1.0F;
 
 /** The observation model Matrix (notation in literature: H). */
 const Eigen::Matrix<float, NUMBER_OF_MEASUREMENTS_M, NUMBER_OF_STATES_N> ExtendedKalmanFilter::OBSERVATION_MATRIX_H{
     {1.0F, 0.0F, 0.0F, 0.0F},
     {0.0F, 1.0F, 0.0F, 0.0F},
+    {0.0F, 0.0F, 1.0F, 0.0F},
     {0.0F, 0.0F, 0.0F, 1.0F}};
 
 /** The covariance of the observation noise Matrix (notation in literature: R). */
 const Eigen::Matrix<float, NUMBER_OF_MEASUREMENTS_M, NUMBER_OF_MEASUREMENTS_M>
-    ExtendedKalmanFilter::OBSERVATION_NOISE_MATRIX_R{{0.0F, 0.0F, 0.0F}, {0.0F, 0.0F, 0.0F}, {0.0F, 0.0F, 0.0F}};
+    ExtendedKalmanFilter::OBSERVATION_NOISE_MATRIX_R{{ODOMETRY_POSITION_VARIANCE, 0.0F, 0.0F, 0.0F},
+                                                     {0.0F, ODOMETRY_POSITION_VARIANCE, 0.0F, 0.0F},
+                                                     {0.0F, 0.0F, ODOMETRY_VELOCITY_VARIANCE, 0.0F},
+                                                     {0.0F, 0.0F, 0.0F, ODOMETRY_ANGLE_VARIANCE}};
+
+/** The covariance matrix of the process noise Matrix (notation in literature: Q). */
+const Eigen::Matrix<float, NUMBER_OF_STATES_N, NUMBER_OF_STATES_N> ExtendedKalmanFilter::PROCESS_COVARIANCE_MATRIX_Q{
+    {0.0F, 0.0F, 0.0F, 0.0F},
+    {0.0F, 0.0F, 0.0F, 0.0F},
+    {0.0F, 0.0F, ACCELERATION_VARIANCE, 0.0F},
+    {0.0F, 0.0F, 0.0F, TURNRATE_VARIANCE}};
+
 /******************************************************************************
  * Public Methods
  *****************************************************************************/
-void ExtendedKalmanFilter::init()
+
+void ExtendedKalmanFilter::init(KalmanParameter& initialParameter)
 {
-    /* TODO: Implement Kalman Filter in cpp (TD072) */
+    m_stateVector(IDX_POSITION_X_STATE_VECTOR)  = initialParameter.positionOdometryX;
+    m_stateVector(IDX_POSITION_Y_STATE_VECTOR)  = initialParameter.positionOdometryY;
+    m_stateVector(IDX_VELOCITY_STATE_VECTOR)    = 0.0F;
+    m_stateVector(IDX_ORIENTATION_STATE_VECTOR) = initialParameter.angleOdometry;
+    updateControlInputVector(initialParameter);
+    updateMeasurementVector(initialParameter);
 }
 
 void ExtendedKalmanFilter::predictionStep(const uint16_t timeStep)
 {
-    /* TODO: Implement Kalman Filter in cpp (TD072) */
+    /* Extract individual values into variables in favor of readability. */
+    float positionX   = m_stateVector(IDX_POSITION_X_STATE_VECTOR);  /* In mm */

```

Comment:
In other files of the repository the units are specified in comments instead of the variable name. I propose to do the same to keep an uniform styling.